test-specifications/token-service/tokenMintTransaction.md

```diff
@@ -0,0 +1,200 @@
+|---------|--------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------|-------------------|
+| 1       | Mints an amount of 1,000,000 fungible tokens                                   | tokenId=<CREATED_FUNGIBLE_TOKEN_ID>, amount="1000000", commonTransactionParams.signers=[<CREATED_FUNGIBLE_TOKEN_SUPPLY_KEY>]              | The token mint succeeds and the token's treasury account contains 1,000,000 more minted tokens.                 | N                 |
+| 2       | Mints an amount of 0 fungible tokens                                           | tokenId=<CREATED_FUNGIBLE_TOKEN_ID>, amount="0", commonTransactionParams.signers=[<CREATED_FUNGIBLE_TOKEN_SUPPLY_KEY>]                    | The token mint succeeds and the token's treasury account contains 0 more minted tokens.                         | N                 |
+| 3       | Mints no fungible tokens                                                       | tokenId=<CREATED_FUNGIBLE_TOKEN_ID>, commonTransactionParams.signers=[<CREATED_FUNGIBLE_TOKEN_SUPPLY_KEY>]                                | The token mint succeeds and the token's treasury account contains 0 more minted tokens.                         | N                 |
+| 4       | Mints an amount of 9,223,372,036,854,775,806 (int64 max - 1) fungible tokens   | tokenId=<CREATED_FUNGIBLE_TOKEN_ID>, amount="9223372036854775806", commonTransactionParams.signers=[<CREATED_FUNGIBLE_TOKEN_SUPPLY_KEY>]  | The token mint succeeds and the token's treasury account contains 9,223,372,036,854,775,806 more minted tokens. | N                 |
+| 5       | Mints an amount of 9,223,372,036,854,775,807 (int64 max) fungible tokens       | tokenId=<CREATED_FUNGIBLE_TOKEN_ID>, amount="9223372036854775807", commonTransactionParams.signers=[<CREATED_FUNGIBLE_TOKEN_SUPPLY_KEY>]  | The token mint succeeds and the token's treasury account contains 9,223,372,036,854,775,807 more minted tokens. | N                 |
+| 6       | Mints an amount of 9,223,372,036,854,775,808 (int64 max + 1) fungible tokens   | tokenId=<CREATED_FUNGIBLE_TOKEN_ID>, amount="9223372036854775808", commonTransactionParams.signers=[<CREATED_FUNGIBLE_TOKEN_SUPPLY_KEY>]  | The token mint fails with an INVALID_TOKEN_MINT_AMOUNT response code from the network.                          | N                 |
+| 7       | Mints an amount of 18,446,744,073,709,551,614 (uint64 max - 1) fungible tokens | tokenId=<CREATED_FUNGIBLE_TOKEN_ID>, amount="18446744073709551614", commonTransactionParams.signers=[<CREATED_FUNGIBLE_TOKEN_SUPPLY_KEY>] | The token mint fails with an INVALID_TOKEN_MINT_AMOUNT response code from the network.                          | N                 |
```

Comment:
> Mints an amount of 18,446,744,073,709,551,614 (uint64 max - 1)

I feel like we would have similar problems with these uints as before.

Reply:
We would have to change the Java API to be `BigInteger`, because now it's `long`. This will most likely break users and I think it's not worth it.

Reply:
Yeah since Java doesn't support unsigned it won't technically support this large number but what we're after is the underlying bit structure. What would happen in Java if `Long.fromString(<uint64 max>)`? Would it just set it to -1? I feel like that'd be fine since the underlying bit structure is the same. If not then some extra parsing may be required

Reply:
`Long.parseLong("18446744073709551614")` results in `NumberFormatException` .  I don't think we can "overflow" the value - java will prevent that from happening.

Reply:
Protocol buffers `protoc` (and PBJ) translate the `uint64` to a java `long` when deserializing.  It does so bitwise, so values above `Long.MAX_VALUE` are just negative `long` values in Java.
This is clearly documented in the proto3 language guide.

Reply:
Alright, that will do it. Thanks @rwalworth @jsync-swirlds 
@ivaylogarnev-limechain Do you think we can do something similar in JS?